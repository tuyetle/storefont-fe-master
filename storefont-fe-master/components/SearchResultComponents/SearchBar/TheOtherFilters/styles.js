import variables from '~/styles/variables';
import { css } from 'glamor';
import themes from '~/styles/themes';

export default {
  filters: css({
    position: 'fixed',
    top: '0',
    right: '0',
    left: '0',
    borderRadius: '0',
    width: '100vw',
    maxWidth: variables.$maxWidthFilter,
    overflow: 'hidden',
    margin: '0',
    backgroundColor: 'white',
    boxShadow: variables.$boxShadowFilter,
    border: `1px solid ${variables.$borderFilter}`,
    color: variables.$colorWelcome,
    padding: '1.5em',
    display: 'none',

    '.closedFilter &': {
      display: 'block',
      zIndex: '9',
      height: '100vh',
      maxHeight: '100vh',
      overflowY: 'scroll',
      overflowX: 'hidden',
      padding: '0',
    },

    '@media (min-width: 768px)': {
      position: 'fixed',
      width: '596px',
      right: '0',
      left: '-18em',
      display: 'block',
    },
  }),
  filter: css({
    padding: '1em',
    '&:nth-child(2n+0)': {
      borderLeft: `1px solid ${variables.$borderdropdown}`,
    },
  }),
  filterList: css({
    margin: '0',
    padding: '0',
    textAlign: 'left',
  }),
  filterListItem: css({
    width: '33%',
    lineHeight: '1em',
    display: 'inline-block',
    verticalAlign: 'middle',

    '& .form-check': {
      marginBottom: '0',
    },

    '@media (min-width: 768px)': {
      width: '50%',
      '& .form-check': {
        marginBottom: '.5rem',
      },
    },
  }),
  filterName: css({
    textAlign: 'left',
    fontSize: '1em',
    fontWeight: 'bold',
    marginBottom: '0.8em',
    '@media (min-width: 768px)': {
      marginBottom: '1.5em',
    },
  }),
  label: css({
    overflow: 'hidden',
    marginBottom: '0',
    '& input[type=checkbox]': {
      width: '1em',
      height: '1em',
      display: 'inline-block',
      verticalAlign: 'middle',
      borderColor: variables.$colorHambuger,
      '&:before': {
        top: '0',
        fontSize: '0.8em',
      },
    },
    '@media (min-width: 768px)': {
      marginBottom: '.5rem',
    },
  }),
  optionsName: css({
    paddingLeft: '0.5em',
    fontSize: '0.8125em',
    lineHeight: '1.65em',
    '@media (min-width: 768px)': {
      fontSize: '0.875em',
    },
  }),
  groupSell: css({
    display: 'inline-block',
    color: variables.$colorWelcome,
    borderLeft: 'none',
    padding: '0',
    height: '41px',
    lineHeight: '2.5em',
    float: 'left',
    marginBottom: '0.8em',
    '& .ant-radio-group-large .ant-radio-button-wrapper': {
      height: '29px',
      lineHeight: '26px',
      '&:before': {
        display: 'none',
      },
    },
    '& .ant-radio-button-wrapper': {
      color: 'black',
      border: '0.0625em solid transparent',
      textTransform: 'uppercase',
      boxShadow: 'none',
      outline: '0',
      borderRadius: '0',
      fontSize: '0.9375em',

      '&:hover': {
        color: variables.$colorWelcome,
        boxShadow: 'none',
      },

      '&:focus, &:active': {
        borderRadius: '0',
        color: variables.$colorWelcome,
        border: `0.0625em solid ${variables.$colorHambuger}`,
        boxShadow: 'none',
        outline: 'none',
      },

      '&.ant-radio-button-wrapper-checked': {
        borderRadius: '0',
        border: `0.0625em solid ${variables.$colorHambuger}`,
        color: variables.$colorWelcome,
        '&:hover': {
          boxShadow: 'none',
        },

        '&:first-child': {
          border: `0.0625em solid ${variables.$colorHambuger}`,
          borderRadius: '0',
        },
        '&:last-child': {
          border: `0.0625em solid ${variables.$colorHambuger}`,
          borderRadius: '0',
        },
      },
    },
    '@media (min-width: 768px)': {
      float: 'none',
      borderLeft: `1px solid ${variables.$borderdropdown}`,
      padding: '0 0.5em',
      marginBottom: '0',
    },
  }),
  filtersControl: css({
    width: '100%',
    padding: '0.6em 0 0.3em',
    overflow: 'hidden',
  }),
  filtersName: css({
    ...themes['font-montserrat-semibold'],
    float: 'left',
  }),
  otherFilters: css({
    float: 'right',
  }),
  filtersControlWrap: css({
    boxShadow: variables.$boxShadowFilter,
  }),
  filtersControlContent: css({
    padding: '1em 0 0 0',
  }),
  categoryFilters: css({
    '&.nav': {
      '& .nav-item': {
        width: '100%',
        '& .nav-link': {
          padding: '1em 0',
          cursor: 'pointer',
        },
      },
    },
  }),
  buttonLink: css({
    width: '100%',
    outline: 'none',
    background: 'none',
    cursor: 'pointer',
    borderBottom: 'none',
    WebkitAppearance: 'none',
    padding: '0.5em 0 1em',
    border: '0',
    position: 'relative',
    textAlign: 'left',
    paddingBottom: '0',
    '&:nth-last-child()': {
      borderBottom: '0',
    },

    '&:active, &:hover, &:focus': {
      outline: 'none',
    },
    '&:after': {
      content: '""',
      position: 'absolute',
      top: '1em',
      right: '0',
      border: `solid ${variables.$listingItemTitleColor}`,
      borderWidth: '0 2px 2px 0',
      display: 'inline-block',
      padding: '5px',
      cursor: 'pointer',
      transform: 'rotate(45deg)',
      WebkitTransform: 'rotate(45deg)',
    },
    '&.active': {
      borderBottom: '0',
      '&:after': {
        top: '1.4em',
        transform: 'rotate(-135deg)',
        WebkitTransform: 'rotate(-135deg)',
      },
    },

  }),

  categoryFiltersContent: css({
    '&.nav': {
      paddingLeft: '1em',
      marginBottom: '0.1em',
      borderBottom: `1px solid ${variables.$borderMenuLink}`,
    },
  }),
};
