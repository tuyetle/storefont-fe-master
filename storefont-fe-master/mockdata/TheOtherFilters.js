const result = {
    categories: [
        { name: 'Nhà phố', value: '1' },
        { name: 'Biệt thự', value: '2' },
        { name: 'Căn hộ', value: '3' },
        { name: 'Căn hộ dịch vụ', value: '4' },
        { name: 'Đất nền', value: '5' },
        { name: 'Văn phòng', value: '6' },
        { name: 'Khác', value: '7' },
    ],

    furnitures: [
        { name: 'Sofa', value: 'sofa' },
        { name: 'Nhà bếp', value: 'kitchen' },
        { name: 'Tivi', value: 'television' },
        { name: 'Máy giặt', value: 'washingmachine' },
        { name: 'Tủ lạnh', value: 'fridge' },
        { name: 'Máy lạnh', value: 'airconditioner' },
        { name: 'Máy nước nóng', value: 'waterheater' },
    ],

    utilities: [
        { name: 'Internet', value: 'internet' },
        { name: 'Truyền hình cáp', value: 'cctv' },
        { name: 'Thang máy', value: 'lift' },
        { name: 'Hồ bơi', value: 'swimmingpool' },
        { name: 'Gym', value: 'gym' },
        { name: 'Mở cửa 24h', value: '247' },
        { name: 'Bãi để ôtô', value: 'otoparking' },
    ],
};

export default result;
