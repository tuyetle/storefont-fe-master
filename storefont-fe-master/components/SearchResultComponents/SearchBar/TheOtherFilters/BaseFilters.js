import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { FormGroup, Label, Row, Col } from 'reactstrap';
import Checkbox from '~/components/common/Checkbox/Checkbox';

import { DIRECTION_FILTER_OPTIONS } from '~/config/config';
import styles from './styles';

function renderCheckboxGroup(group, label, options, selectedValues, onChange) {
  const selected = selectedValues[group] || [];

  return (
    <Col className={classNames(`${styles.filter}`, 'col-12')} sm="6">
      <h4 className={`${styles.filterName}`}>{label}</h4>
      <ul className={classNames(`${styles.filterList}`, 'list-unstyled', 'd-flex', 'align-items-stretch', 'flex-wrap')}>
        {options.map(opt => (
          <li key={opt.name} className={`${styles.filterListItem}`}>
            <FormGroup check>
              <Label className={`${styles.label}`}>
                <Checkbox
                  type="checkbox"
                  value={opt.value}
                  name={group}
                  checked={selected.indexOf(opt.value) >= 0}
                  onChange={onChange}
                />
                <span className={`${styles.optionsName}`}>
                  {opt.name}
                </span>
              </Label>
            </FormGroup>
          </li>
        ))}
      </ul>
    </Col>
  );
}

const BaseFilters = ({
  categories, utilities, furnitures, t,
  selectedValues, onChange,
}) => (
  <Row>
    {renderCheckboxGroup('categoryIds', t('searchResult:categoryFilter'), categories, selectedValues, onChange)}
    {renderCheckboxGroup(
      'directions', t('searchResult:directionFilter'),
      DIRECTION_FILTER_OPTIONS.map(opt => ({ name: t(`searchResult:${opt.name}`), value: opt.name })),
      selectedValues, onChange,
    )}
    {renderCheckboxGroup('utilities', t('searchResult:utilityFilter'), utilities, selectedValues, onChange)}
    {renderCheckboxGroup('furnitures', t('searchResult:furnitureFilter'), furnitures, selectedValues, onChange)}
  </Row>
);

BaseFilters.propTypes = {
  onChange: PropTypes.func.isRequired,
  selectedValues: PropTypes.any,
  categories: PropTypes.array.isRequired,
  furnitures: PropTypes.array.isRequired,
  utilities: PropTypes.array.isRequired,
  t: PropTypes.func.isRequired,
};

BaseFilters.defaultProps = {
  selectedValues: {
    categoryIds: [],
    directions: [],
    furnitures: [],
    utilities: [],
  },
};

export default BaseFilters;
