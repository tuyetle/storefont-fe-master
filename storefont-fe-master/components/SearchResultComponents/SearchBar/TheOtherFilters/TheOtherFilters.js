import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { cloneDeep } from 'lodash';
import BaseFilters from './BaseFilters';

class TheOtherFilters extends Component {
  onChange = (evt) => {
    const {
      fieldApi: { value, setValue, setTouched },
    } = this.props;
    const newValue = cloneDeep(value);
    if (evt.target.type === 'radio') {
      newValue[evt.target.name] = evt.target.value;
    } else if (evt.target.type === 'checkbox') {
      if (!newValue[evt.target.name]) newValue[evt.target.name] = [];

      if (evt.target.checked === true) {
        newValue[evt.target.name].push(evt.target.value);
      } else {
        newValue[evt.target.name].splice(newValue[evt.target.name].indexOf(evt.target.value), 1);
      }
    }

    setTouched(true);
    setValue(newValue);
  }

  render() {
    const {
      fieldApi: { value },
      t, categories, furnitures, utilities,
    } = this.props;

    const selectedValues = { };
    if (value) {
      selectedValues.categoryIds = value.categoryIds || [];
      selectedValues.directions = value.directions || [];
      selectedValues.furnitures = value.furnitures || [];
      selectedValues.utilities = value.utilities || [];
    }

    return (
      <BaseFilters
        onChange={this.onChange}
        categories={categories}
        furnitures={furnitures}
        utilities={utilities}
        selectedValues={selectedValues}
        t={t}
      />
    );
  }
}

TheOtherFilters.propTypes = {
  fieldApi: PropTypes.object.isRequired,
  categories: PropTypes.array.isRequired,
  furnitures: PropTypes.array.isRequired,
  utilities: PropTypes.array.isRequired,
  t: PropTypes.func.isRequired,
};

TheOtherFilters.defaultProps = {
};

export default TheOtherFilters;
