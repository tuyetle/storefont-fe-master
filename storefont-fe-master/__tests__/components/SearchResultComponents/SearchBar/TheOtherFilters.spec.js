import React from 'react';
import renderer from 'react-test-renderer';
import TheOtherFilters from '~/components/SearchResultComponents/SearchBar/TheOtherFilters/TheOtherFilters';

describe('(Component) TheOtherFilters', () => {
  let component;
  let props;
  beforeEach(() => {
    props = {
      fieldApi: {},
      categories: [{
        name: 'House',
        value: '1',
      }],
      furnitures: [{
        name: 'Sofa',
        value: '2',
      }],
      utilities: [{
        name: 'Wifi',
        value: '3',
      }],
      t: () => {},
    };

    component = <TheOtherFilters {...props} />;
  });

  it('should render conrectly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

