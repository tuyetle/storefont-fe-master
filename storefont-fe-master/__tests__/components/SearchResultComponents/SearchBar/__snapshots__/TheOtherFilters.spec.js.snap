// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) TheOtherFilters should render conrectly 1`] = `
<div
  className="row"
>
  <div
    className="css-1wzb0xk col-12 col col-sm-6"
  >
    <h4
      className="css-o1j11f"
    />
    <ul
      className="css-11q3crl list-unstyled d-flex align-items-stretch flex-wrap"
    >
      <li
        className="css-6hsheu"
      >
        <div
          className="form-check"
        >
          <label
            className="css-2d34dm "
            htmlFor={undefined}
          >
            <input
              checked={false}
              className="css-1k7inrl"
              name="categoryIds"
              onChange={[Function]}
              type="checkbox"
              value="1"
            />
            <span
              className="css-1qgrpm7"
            >
              House
            </span>
          </label>
        </div>
      </li>
    </ul>
  </div>
  <div
    className="css-1wzb0xk col-12 col col-sm-6"
  >
    <h4
      className="css-o1j11f"
    />
    <ul
      className="css-11q3crl list-unstyled d-flex align-items-stretch flex-wrap"
    >
      <li
        className="css-6hsheu"
      >
        <div
          className="form-check"
        >
          <label
            className="css-2d34dm "
            htmlFor={undefined}
          >
            <input
              checked={false}
              className="css-1k7inrl"
              name="directions"
              onChange={[Function]}
              type="checkbox"
              value="east"
            />
            <span
              className="css-1qgrpm7"
            />
          </label>
        </div>
      </li>
      <li
        className="css-6hsheu"
      >
        <div
          className="form-check"
        >
          <label
            className="css-2d34dm "
            htmlFor={undefined}
          >
            <input
              checked={false}
              className="css-1k7inrl"
              name="directions"
              onChange={[Function]}
              type="checkbox"
              value="west"
            />
            <span
              className="css-1qgrpm7"
            />
          </label>
        </div>
      </li>
      <li
        className="css-6hsheu"
      >
        <div
          className="form-check"
        >
          <label
            className="css-2d34dm "
            htmlFor={undefined}
          >
            <input
              checked={false}
              className="css-1k7inrl"
              name="directions"
              onChange={[Function]}
              type="checkbox"
              value="south"
            />
            <span
              className="css-1qgrpm7"
            />
          </label>
        </div>
      </li>
      <li
        className="css-6hsheu"
      >
        <div
          className="form-check"
        >
          <label
            className="css-2d34dm "
            htmlFor={undefined}
          >
            <input
              checked={false}
              className="css-1k7inrl"
              name="directions"
              onChange={[Function]}
              type="checkbox"
              value="north"
            />
            <span
              className="css-1qgrpm7"
            />
          </label>
        </div>
      </li>
      <li
        className="css-6hsheu"
      >
        <div
          className="form-check"
        >
          <label
            className="css-2d34dm "
            htmlFor={undefined}
          >
            <input
              checked={false}
              className="css-1k7inrl"
              name="directions"
              onChange={[Function]}
              type="checkbox"
              value="northEast"
            />
            <span
              className="css-1qgrpm7"
            />
          </label>
        </div>
      </li>
      <li
        className="css-6hsheu"
      >
        <div
          className="form-check"
        >
          <label
            className="css-2d34dm "
            htmlFor={undefined}
          >
            <input
              checked={false}
              className="css-1k7inrl"
              name="directions"
              onChange={[Function]}
              type="checkbox"
              value="northWest"
            />
            <span
              className="css-1qgrpm7"
            />
          </label>
        </div>
      </li>
      <li
        className="css-6hsheu"
      >
        <div
          className="form-check"
        >
          <label
            className="css-2d34dm "
            htmlFor={undefined}
          >
            <input
              checked={false}
              className="css-1k7inrl"
              name="directions"
              onChange={[Function]}
              type="checkbox"
              value="southEast"
            />
            <span
              className="css-1qgrpm7"
            />
          </label>
        </div>
      </li>
      <li
        className="css-6hsheu"
      >
        <div
          className="form-check"
        >
          <label
            className="css-2d34dm "
            htmlFor={undefined}
          >
            <input
              checked={false}
              className="css-1k7inrl"
              name="directions"
              onChange={[Function]}
              type="checkbox"
              value="southWest"
            />
            <span
              className="css-1qgrpm7"
            />
          </label>
        </div>
      </li>
    </ul>
  </div>
  <div
    className="css-1wzb0xk col-12 col col-sm-6"
  >
    <h4
      className="css-o1j11f"
    />
    <ul
      className="css-11q3crl list-unstyled d-flex align-items-stretch flex-wrap"
    >
      <li
        className="css-6hsheu"
      >
        <div
          className="form-check"
        >
          <label
            className="css-2d34dm "
            htmlFor={undefined}
          >
            <input
              checked={false}
              className="css-1k7inrl"
              name="utilities"
              onChange={[Function]}
              type="checkbox"
              value="3"
            />
            <span
              className="css-1qgrpm7"
            >
              Wifi
            </span>
          </label>
        </div>
      </li>
    </ul>
  </div>
  <div
    className="css-1wzb0xk col-12 col col-sm-6"
  >
    <h4
      className="css-o1j11f"
    />
    <ul
      className="css-11q3crl list-unstyled d-flex align-items-stretch flex-wrap"
    >
      <li
        className="css-6hsheu"
      >
        <div
          className="form-check"
        >
          <label
            className="css-2d34dm "
            htmlFor={undefined}
          >
            <input
              checked={false}
              className="css-1k7inrl"
              name="furnitures"
              onChange={[Function]}
              type="checkbox"
              value="2"
            />
            <span
              className="css-1qgrpm7"
            >
              Sofa
            </span>
          </label>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
